File: acttest/Actuarialvaluations/clientapp/src/models/common/MarkovStateDropdown.ts
/**
 * The Markov State Dropdown model class
 */
export class MarkovStateDropdown {
    
    public StateId: number | null = null;    
    public StateName: string | null = null;    
    public StateCode: string | null = null;   
    public StartDate: Date | null = null;   
    public EndDate: Date | null = null;
    public Selected: boolean | null = false;
    
    constructor(data: any = null) {
        if (data) {
            Object.assign(this, data);
        }
    }    
}

File: acttest/Actuarialvaluations.Models/API/MarkovStateDropdown.cs
namespace Actuarialvaluations.Models.API
{
  using System;
  using Newtonsoft.Json;

  /// <summary>
  /// The Markov state class
  /// </summary>
  public class MarkovStateDropdown : Entity
  {
    /// <summary>
    /// Gets or sets StateId
    /// </summary>
    [JsonProperty("StateId")]
    public int StateId { get; set; }

    /// <summary>
    /// Gets or sets StateName
    /// </summary>
    [JsonProperty("StateName")]
    public string StateName { get; set; }

    /// <summary>
    /// Gets or sets StateCode
    /// </summary>
    [JsonProperty("StateCode")]
    public string StateCode { get; set; }

    /// <summary>
    /// Gets or sets StartDate
    /// </summary>
    [JsonProperty("StartDate")]
    public DateTime StartDate { get; set; }

    /// <summary>
    /// Gets or sets EndDate
    /// </summary>
    [JsonProperty("EndDate")]
    public DateTime EndDate { get; set; }

    /// <summary>
    /// Gets or sets Selected
    /// </summary>
    [JsonProperty("Selected")]
    public bool Selected { get; set; }
  }
}
